experimental/framework/fluid-static/src/serviceAudience.ts

```diff
@@ -0,0 +1,167 @@
+    // the session
+    const lastEditedMemberResults = this.getLastEdited();
+    if (lastEditedMemberResults !== undefined) {
+      this.emit("lastEditedChanged", lastEditedMemberResults);
+    }
+  }
+
+  /**
+   * @inheritdoc
+   */
+  public getMembers(): Map<string, IMember> {
+    const users = new Map<string, IMember>();
+    // Iterate through the members and get the user specifics.
+    this.audience.getMembers().forEach((member: IClient, clientId: string) => {
+      // Get all the current human members
+      if (member.details.capabilities.interactive) {
+        const userId = member.user.id;
+        if (users.has(userId)) {
+          const existingValue = users.get(userId);
+          if (existingValue) {
+            existingValue.connectedClients.push({
+              clientId,
+              connectionMode: member.mode,
+              timeLastActive: this.lastEditedTimesByClient.get(clientId),
+            });
+            existingValue.connectedClients.sort((a, b) =>
+              (b.timeLastActive?.getMilliseconds() ?? 0) - (a.timeLastActive?.getMilliseconds() ?? 0));
+          }
+        } else {
+          users.set(userId, {
+            userId,
+            connectedClients: [{
+              clientId,
+              connectionMode: member.mode,
+              timeLastActive: this.lastEditedTimesByClient.get(clientId),
+            }],
+          });
+        }
+      }
+    });
+    return users;
+  }
+
+  /**
+   * @inheritdoc
+   */
+  public getCurrentClient(): IConnectedClient | undefined {
+    const clientId = this.container.clientId;
+    if (clientId === undefined) {
+      return undefined;
+    }
+    const client = this.audience.getMember(clientId);
+    if (client === undefined) {
+      throw Error(`Failed to find client ${clientId} even after it is connected`);
+    }
+    return {
+      clientId,
+      connectionMode: client.mode,
+      timeLastActive: this.lastEditedTimesByClient.get(clientId),
+    };
+  }
+
+  /**
+   * @inheritdoc
+   */
+  public getCurrentMember(): IMember | undefined {
+    const clientId = this.container.clientId;
+    if (clientId === undefined) {
+      return undefined;
+    }
+    return this.getMemberByClientId(clientId);
+  }
+
+  /**
+   * @inheritdoc
+   */
+  public getLastEdited(): ILastEditedResult<IMember> | undefined {
+    const lastEditDetails = this.lastEditedTracker?.getLastEditDetails();
+    if (lastEditDetails !== undefined) {
+      const timestamp = new Date(lastEditDetails.timestamp);
+      this.lastEditedTimesByClient.set(lastEditDetails.clientId, timestamp);
+      const userId = lastEditDetails.user.id;
+      const member: IMember = {
+        userId,
+        connectedClients: this.getMembers().get(userId)?.connectedClients ?? [],
+      };
+      return { member, timestamp };
+    }
+  }
+
+  protected getMemberByClientId(clientId: string): IMember | undefined {
```

Comment:
Looks like this could be private (at least for now, and can always be opened up more in the future)?

Reply:
Converted to private